### PR TITLE
Ternary conditionals

### PR DESCRIPTION
**Pull request by drewnoakes**
Not for everyone, but I like it.

This PR uses the ternary conditional operation (`?:`) in a bunch of places.

I find it reduces repetition and improves readability, but it is subjective. I'm not hugely attached to this PR, but I do prefer the change.

What did I do to test the code and ensure quality:
 - Manual review
 - Manual testing
 - Unit tests

Has been tested on:
 - GIT 2.16
 - Windows 10

### CODE REVIEW

File: GitCommands/Config/ConfigFile.cs [hunk 1]
```diff
@@ -507,15 +507,9 @@ private ParsePart ReadValue(char c)
...
-                                _token.Append(_valueToken);
-                            }
-                            else
-                            {
-                                _token.Append(_valueToken.ToString().Trim());
-                            }
-
+                            _token.Append(_quotedValue
+                                ? _valueToken.ToString()
+                                : _valueToken.ToString().Trim());
```
Comment (RussKie) [inline GitCommands/Config/ConfigFile.cs]: This is some strange piece of code

Comment (drewnoakes) [reply GitCommands/Config/ConfigFile.cs]: It does look a bit strange. These kinds of state-based lexers are very powerful but can be hard to reason about. Gotta get in the zone. And have lots of tests.

**Comment by vbjay** [comment]
I actually prefer it. It is very clear what it's doing. No scrolling is
required. You should understand a ternary. I actually like the values on
separate lines.

**Comment by drewnoakes** [comment]
Rebased to fix conflicts.

**Comment by gerhardol** [comment]
@drewnoakes Many PRs open now. Since @RussKie has approved some of yours as well as #4601, I suggest you merge them with your preferred strategy if you believe they are ready.

**Comment by drewnoakes** [comment]
@gerhardol would you or perhaps @RussKie please explain the process for merging in this repo? Can authors merge their own PRs? Do PRs require a certain number of acceptances? What's the significance of the 'ready' tag? Thanks.

I'll hold off on merging many of these until #4601 goes in. There'll probably be conflicts, and I'm happy to take the hit where needed.

**Comment by gerhardol** [comment]
@drewnoakes I have been a maintainer just slightly longer than you and have certainly less experience in gui programming so you are the seniority here. 

There is no defined process, similar to the merge strategy, one is needed. @russkie is the architecht if he says it is ready or you have reason to believe he will approve it, then merge the prs. 

For #4601 I want you to have your say too, I will not find any real to remark on in a review.

**Comment by RussKie** [comment]
I mark PRs,  which I looked at and believe are ready as such ('ready' tag).
If noone else has comments or feedback for those - feel free to merge them.
Please untag them once merged.

If a PR fixes an issue - please make sure the issue is added to a milestone.
If PR isn't attached to an issue - please add the PR to a milestone.
The next milestone is 3.0. If a bug needs to be backpatched - 3.0+2.5x
milestone.

I hope this makes sense.

I'll add the above to the maintainer guidelines, when able.

**Comment by drewnoakes** [comment]
> I have been a maintainer just slightly longer than you

Oh nice, I didn't realise. You seem very familiar with everything 👍 

@RussKie thank you for the clarification. I'm sure there's more to say on the topic in time. For now I tried to capture your instructions on [the wiki here](https://github.com/gitextensions/gitextensions/wiki/Guidelines:-maintainer-conduct). Please review when you have a chance to make sure I haven't misquoted you. For example, did you mean that only you would tag PRs as `Ready`?

> RussKie is the architecht if he says it is ready or you have reason to believe he will approve it, then merge the prs.

> I mark PRs, which I looked at and believe are ready as such ('ready' tag). If noone else has comments or feedback for those - feel free to merge them.

@RussKie do you only want PRs merged that you've marked as ready, or is that more burden than you care for. Understanding your expectation here would help me contribute more in this area.

**Comment by RussKie** [comment]
> For example, did you mean that only you would tag PRs as Ready?

I have been promoted (more or less) to the product owner role, and as per agile ceremonies it befallen unto me to mark items as ready. :)
Any member can certainly triage and apply tags, but this is one of the tags we don't want to abuse.

I created the tag to make it a bit more obvious when something has been reviewed and is ready to go in, but may not necessarily be merged immediately - e.g. your PRs were good to go but were queued up before the Sam's refactor.

**Comment by RussKie** [comment]
> do you only want PRs merged that you've marked as ready, or is that more burden than you care for. 

As per above - for me it is an indication that I'm happy for the PR to go in. 
With so many active streams there are contentions, and not everything can be merged immediately without causing unnecessary pain to others.
Smaller PRs can typically be merged in without needing a tag.

**Pull request merged**